### DISCUSSION TITLE
Magic-URL authentication

### DISCUSSION THREAD

**Discussion by jamesnottidge** [appwrite/appwrite#5286] [Q&A]
I'm attempting to authenticate users(log-in existing users) via magic-URL.
In order to do that, I'd need to pass in the users ID and email address. 
The issue with this is: my userID is not their email address. 

I have a separate unique ID. 
As such, without logging the user in, I can see no way of obtaining said ID. 

Please can anyone help me with a way forward?

**Comment by stnguyen90** [comment]
@jamesnottidge, as the [docs](https://appwrite.io/docs/client/account?sdk=web-default#accountCreateMagicURLSession) describe, the userId parameter is used if an account with the supplied email doesn't exist yet to create an account. If the account already exists, the userId parameter is ignored.

**Comment by jamesnottidge** [comment]
Thank you @stnguyen90, however, the docs don’t clearly describe this. Infact, the docs say:

"If the provided user ID has not be registered, a new user will be created"

Hence, my confusion.  Thank you, for shedding light on this for me. 

Should I open an issue to state this in less ambiguous terms?

**Reply by stnguyen90** [reply]
> Should I open an issue to state this in less ambiguous terms?

Sure.

Btw, GitHub Discussions has threads so make sure to reply to the existing thread so that it's easier to follow a conversation.

**Reply by jamesnottidge** [reply]
Thank you!
I'd do that on the docs now.
